version: "3"
services:
  auth_server:
    build:
      context: ./
      dockerfile: dockerfile
    ports:
      - "5001:5001"
    restart:
      on-failure
    environment:
      - spring.datasource.url=jdbc:mysql://mysqldb:3306/authdb?useSSL=false&allowPublicKeyRetrieval=true&createDatabaseIfNotExist=true
    networks:
      - network
    depends_on:
      mysqldb:
        condition: service_healthy

  mysqldb:
    image: mysql:latest
    networks:
      - network
    volumes:
      - ./initDB.sql:/docker-entrypoint-initdb.d/initDB.sql
    environment:
      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
      - MYSQL_DATABASE=authdb
    healthcheck:
      test: [ "CMD", "mysqladmin" ,"ping", "-h", "localhost" ]
      timeout: 30s
      retries: 10

networks:
  network: